### ISSUE TITLE
Progress?

### ISSUE THREAD

**Issue by garrettw** [TomBZombie/php-dependency-injection-benchmarks#15]
Any chance v2 will see the light of day? I'm sure I'm not the only one who eagerly awaits the next iteration of your analysis, especially with the improvements that have been made to Dice over the past 9-10 months.

(Incidentally, I would like to add that I just ran your benchmark with your latest Dice 2.0 code as well as my own, and mine actually came out a bit ahead on tests 2, 3, and 5. Granted, I did have to increase the number of loop iterations from 10k to 100k, 50k, and 500k, respectively, in order for a clear winner to emerge.  I didn't run test 1, and test 4 was a draw with 100k iterations. I'm not entirely sure of the reason for this outcome yet, but maybe I'll figure it out.)

**Comment by TomBZombie** [comment]
I did add several others including Pimple and Nette as well as another test. Nette is fastest but it has a huge startup overhead. When this is accounted for, it's far far slower than Dice or Pimple.

Interestingly I see your latest commit is "added a bunch of ugly backslashes for speed" does this have much of an advantage?

**Comment by garrettw** [comment]
I did my speed test before that commit, so not sure. I just read about it somewhere the other day and thought I'd try it. I'll let you know.

**Comment by TomBZombie** [comment]
I think I found the cause of the difference. After I added `'call' => []'` to the default rule, the `isset($rule['call'])` check in `getClosure` was always evaluating to true causing each object creation to run both the call version of the closure and the normal version. Try with the latest commit.

I had a play with backslashes but the number of times the global functions actually get called during the benchmarks I have is so small it doesn't affect it.
